device/cuda/src/seeding/seed_finding.cu

```diff
@@ -162,28 +172,42 @@ vecmem::data::vector_buffer<seed> seed_finding::operator()(
     const sp_grid_const_view& g2_view,
     const std::vector<unsigned int>& grid_sizes) const {
 
-    // Get the prefix sum for the spacepoint grid using buffer.
-    const device::prefix_sum_t sp_grid_prefix_sum = device::get_prefix_sum(
-        grid_sizes, (m_mr.host ? *(m_mr.host) : m_mr.main));
+    // Create prefix sum of grid_sizes vector
+    std::vector<unsigned int> grid_sizes_sum(grid_sizes.size());
+    std::partial_sum(grid_sizes.begin(), grid_sizes.end(),
+                     grid_sizes_sum.begin(), std::plus<unsigned int>());
+    const unsigned int& totalGrid = grid_sizes_sum[grid_sizes_sum.size() - 1];
```

Comment:
Let's just make it a `const unsigned int`. For primitive types there's no upside to using references. (A reference is 64 bits, and needs to be followed through once indirection in memory. While `unsigned int` is just 32 bits, and is "right there".)